lib/tapioca/compilers/dsl/active_storage.rb

```diff
@@ -0,0 +1,66 @@
+# typed: strict
+# frozen_string_literal: true
+
+begin
+  require "active_record"
+  require "active_storage"
+  require "active_storage/attached"
+rescue LoadError
+  return
+end
+
+module Tapioca
+  module Compilers
+    module Dsl
+      # `Tapioca::Compilers::Dsl::ActiveStorage` decorates RBI files for subclasses of
+      # `ActiveRecord::Base` that declare [one](https://edgeguides.rubyonrails.org/active_storage_overview.html#has-one-attached)
+      # or [many](https://edgeguides.rubyonrails.org/active_storage_overview.html#has-many-attached) attachments.
+      #
+      # For example, with the following `ActiveRecord::Base` subclass:
+      #
+      # ~~~rb
+      # class Post < ApplicationRecord
+      #  has_one_attached :photo
+      #  has_many_attached :blogs
+      # end
+      # ~~~
+      #
+      # this generator will produce the RBI file `post.rbi` with the following content:
+      #
+      # ~~~rbi
+      # # typed: true
+      # class Post
+      #  def photo; end
+      #  def photo=; end
+      #  def blogs; end
+      #  def blogs=; end
+      # end
+      # ~~~
+      class ActiveStorage < Base
+        extend T::Sig
+
+        sig { override.params(root: RBI::Tree, constant: T.class_of(::ActiveRecord::Base)).void }
+        def decorate(root, constant)
+          return if constant.reflections.empty?
+
+          root.create_path(constant) do |scope|
+            constant.attachment_reflections.each do |field_name, _value|
```

Comment:
The `value` parameter here should be used since it tells us if this is a one or many attachment:
```ruby
class User < ActiveRecord::Base
  has_one_attached :foo
  has_many_attached :bar
end

User.attachment_reflections
=> {"foo"=>
  #<ActiveStorage::Reflection::HasOneAttachedReflection:0x00007fb9d616d5b8
   @active_record=User (call 'User.connection' to establish a connection),
   @klass=nil,
   @name=:foo,
   @options={:dependent=>:purge_later, :service_name=>nil},
   @plural_name="foos",
   @scope=nil>,
 "bar"=>
  #<ActiveStorage::Reflection::HasManyAttachedReflection:0x00007fb9d2c2dc90
   @active_record=User (call 'User.connection' to establish a connection),
   @klass=nil,
   @name=:bar,
   @options={:dependent=>:purge_later, :service_name=>nil},
   @plural_name="bars",
   @scope=nil>}
```

Note how the class of the value tells us which kind it is.

Reply:
Actually, I think it is better to use `reflect_on_all_attachments` here instead, since it is a method defined on `ActiveStorage::Reflection::ActiveRecordExtensions::ClassMethods`: http://api.rubyonrails.org/classes/ActiveStorage/Reflection/ActiveRecordExtensions/ClassMethods.html#method-i-reflect_on_all_attachments
```suggestion
            constant.reflect_on_all_attachments.each do |reflection|
```